subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -795,8 +795,14 @@ static void response_cb(struct http_response *rsp,
 		if (hb_context.dl.http_content_size == 0) {
 			body_data = rsp->body_start;
 			body_len = rsp->data_len;
+			// subtract the size of the HTTP header from body_len
 			body_len -= (rsp->body_start - rsp->recv_buf);
 			hb_context.dl.http_content_size = rsp->content_length;
+		} else {
+			// more general case where body data is set, but no need to take the
+			// HTTP header into account
+			body_data = rsp->body_start;
```

Comment:
if the receive buffer (body_start) and data_len is handled here means, 
then the next if statement is not required,
````            
 if ((rsp->body_found == 1) && (body_data == NULL)) {
            body_data = rsp->recv_buf;
            body_len = rsp->data_len;
}
````

Reply:
The if statement is still required. There is still a chance that `rsp->body_start` is set to `NULL`. This is the case when the data to be read starts at rsp->recv_buffer. The code that receives data from the from the socket ([`http_wait_data()`](https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/net/lib/http/http_client.c#L411) doesn't necessarily place the received data at the start of the buffer.